### DISCUSSION TITLE
How to limit vector memory usage

### DISCUSSION THREAD

**Discussion by z2665** [vectordotdev/vector#17942] [Q&A] [answered]
We will use vector as deamonset deployed on each node node for collecting logs.Since the node itself will deploy the service, vector agent will allocate only 500MiB of memory.
But when we process the logs in large amount, vector will request more than 500MiB of memory and cause oom kill.We We have activated the sink disk buffer, but the memory still exceeds the limit, how should we configure the memory limit in vector and avoid the oom?

**Comment by jszwedko** [comment, marked as answer]
Hi @z2665 !

Unfortunately there isn't a way to have Vector automatically apply back pressure when hitting resource constraints yet. https://github.com/vectordotdev/vector/issues/11942 is tracking this feature.
